File: LeetCode/Python/Easy/997-Find-The-Town-Judge/solution.py
class Solution:
    def findJudge(self, n: int, trust: List[List[int]]) -> int:
        result: int=-1

        if len(trust)==0:
            return n if n==1 else -1

        indegree: list[int]=[0] * (n + 1)

        outdegree: list[int]=[0] * (n + 1)

        for edge in trust:
            a: int=edge[0]

            b: int=edge[1]

            outdegree[a]+=1

            indegree[b]+=1

        for person in range(1, n + 1):
            if indegree[person]==n-1 and outdegree[person]==0:
                result=person

                break

        return result

File: LeetCode/Rust/Easy/997-Find-the-Town-Judge/solution.rs
impl Solution
{
    pub fn find_judge(n: i32, trust: Vec<Vec<i32>>) -> i32
    {
        let mut result=-1;

        let mut indegree: Vec<i32>=vec![0; (n + 1) as usize];

        let mut outdegree: Vec<i32>=vec![0; (n + 1) as usize];

        for edge in trust.iter()
        {
            let a: i32=edge[0];

            let b: i32=edge[1];

            outdegree[a as usize]+=1;

            indegree[b as usize]+=1;
        }

        for person in 1..n + 1
        {
            if outdegree[person as usize]==0 && indegree[person as usize]==n-1
            {
                result=person;

                break;
            }
        }

        result
    }
}